lib/Bio/EnsEMBL/DataCheck/Checks/SupertreeTopology.pm

```diff
@@ -0,0 +1,166 @@
+=head1 LICENSE
+
+Copyright [2018-2024] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::SupertreeTopology;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'SupertreeTopology',
+  DESCRIPTION    => 'Check that the topology of each supertree is as expected',
+  GROUPS         => ['compara', 'compara_gene_tree_pipelines', 'compara_gene_trees'],
+  DATACHECK_TYPE => 'critical',
+  DB_TYPES       => ['compara'],
+  TABLES         => ['gene_tree_node', 'gene_tree_root']
+};
+
+
+sub tests {
+  my ($self) = @_;
+
+  my $dbc = $self->dba->dbc;
+  my $helper = $dbc->sql_helper;
+
+  my @supertree_constraints = (
+    "Every node in supertree must have at least one child",
+    "Root of supertree must have at least two children",
+    "Internal nodes of supertree must have two children",
+    "Supertree cannot have invalid subtree links",
+    "Supertree must have at least two subtrees",
+  );
+
+  my $supertree_sql = qq/
+    SELECT
+      root_id
+    FROM
+      gene_tree_root
+    WHERE
+      tree_type = 'supertree'
+  /;
+
+  my $supertree_root_ids = $helper->execute_simple( -SQL => $supertree_sql );
+
+  SKIP: {
+    skip sprintf("There are no supertrees in %s", $dbc->dbname) unless scalar(@{$supertree_root_ids}) > 0;
```

Comment:
skip sprintf("There are no supertrees in %s", $dbc->dbname) if scalar(@{$supertree_root_ids}) == 0;